pipelines/build/common/config_regeneration.groovy

```diff
@@ -0,0 +1,129 @@
+limitations under the License.
+*/
+
+/**
+* This file is a job that regenerates all of the build configurations in pipelines/build/openjdk*_pipeline.groovy to ensure that race
+* conditions are avoided for concurrent builds.
+*
+* This:
+* 1) Is triggered by a githook from openjdk-build whenever there is a push to the master branch
+* 2) Checks if there are any pipelines in progress. If so, this job will run once they have finished and will force other pipelines to  * await for its completion before executing
+* 3) Regenerates the all the job configurations to so they are all matched and in sync with each other. This will avoid one job
+* configuration overwritting another.
+*/
+
+node ("master") {
+    context.stage("Check") {
+        // Download jenkins helper
+        def JobHelper = context.library(identifier: 'openjdk-jenkins-helper@master').JobHelper
+
+        // Get all pipelines (use jenkins api)
+        def parser = new JsonSlurper()
+        def pipelines = []
+
+        try {
+            def get = new URL("https://ci.adoptopenjdk.net/job/build-scripts/api/json?tree=jobs[name]&pretty=true&depth1").openConnection()
+            def rc = get.getResponseCode()
+            def response = parser.parseText(get.getInputStream().getText())
+
+            // Parse api response
+            response.jobs.name.each{ job -> 
+                if (job.contains("pipeline")) {
+                    pipelines.add(job)
+                }
+            }
+
+        } catch (Exception e) {
+            // Failed to connect to jenkins api or a parsing error occured
+            println "Failure on jenkins api connection or parsing. API response code: ${rc}\nError: ${e.getLocalizedMessage()}"
+            currentBuild.result = 'FAILURE'
+        } 
+        finally {
+            // Query jobIsRunning jenkins helper for each pipeline
+            pipelines.each { pipeline ->
+                def inprogress = true
+
+                while (inprogress) {
+                    println "Checking if ${pipeline} is running..."
+                    if (JobHelper.jobIsRunning(pipeline as String)) {
+                        println "${pipeline} is running. Sleeping..."
+                        sleep(900) // sleep for 15mins
+                    }
+                    else {
+                        inprogress = false                    
+                    }
+                }
+            }
+            // No pipelines running or queued up
+            println "No piplines running or scheduled. Running regeneration job..."
+        }
+    }
+
+    context.stage("Regenerate") {
+        // Download openjdk-build
+        def Build = context.library(identifier: 'openjdk-build@master').Build
+
+        // Get all pipelines (use jenkins api)
+        def parser = new JsonSlurper()
+        def pipelines = []
+
+        try {
+            def get = new URL("https://ci.adoptopenjdk.net/job/build-scripts/api/json?tree=jobs[name]&pretty=true&depth1").openConnection()
```

Comment:
Seem like some DRY code here?

Reply:
I added this in as a placeholder for now. I plan on moving the api call to the jenkins helper so it can be accessed by other jobs in the future